x/ibc/07-tendermint/types/update.go

```diff
@@ -115,17 +107,17 @@ func checkValidity(
 	}
 
 	// assert header height is newer than consensus state
-	if header.GetHeight().LTE(consState.Height) {
+	if header.GetHeight().LTE(header.TrustedHeight) {
```

Comment:
This check is unnecessary since it gets checked in `header.ValidateBasic`

Reply:
should we delete it then?

Reply:
According to `header.ValidateBasic` comment:
 ```
// TrustedHeight is less than Header for updates
// and less than or equal to Header for misbehaviour
``` 

Currently the check fails if:

`h.TrustedHeight.GT(h.GetHeight())`

this covers misbehaviour, but equal heights would work for updates. I think the above check should be modified to be `LT` instead of `LTE`?

Reply:
According to `header.ValidateBasic` comment:
 ```
// TrustedHeight is less than Header for updates
// and less than or equal to Header for misbehaviour
``` 

Currently the check fails if:

`h.TrustedHeight.GT(h.GetHeight())`

this covers misbehaviour, but equal heights would work for updates. I think the above check is necessary?